Sources/Plot/API/HTMLComponents.swift

```diff
@@ -579,6 +579,96 @@ extension List: ComponentContainer where Items == ComponentGroup {
     }
 }
 
+// Component used to render an `<picture>` element for responsive image support.
+public struct Picture: Component {
+    /// Type used to define an  Picture source, which points to an image file (or set of them) and associated media queries (and other paramaters).
+    public struct Source {
+        /// image or images  to use - can have parameters
+        public var srcset: String
+        /// sizes for the `srcset` to use - can have parameters
+        public var sizes: String?
+        ///  Media querry (optional)
+        public var media: String?
+        ///  Media type (optional)
+        public var type: String?
+        ///  Intrinsic height of image in pixels - integer with no unit (optional)
+        public var height: Int?
+        ///  Intrinsic width of image in pixels - integer with no unit (optional)
+        public var width: Int?
+
+        /// Initialize a new source.
+        /// - parameter srcset: The set of sources to which this element should point.
+        /// - parameter sizes: Comma separated list of media queries for the `srcset` images.
+        /// - parameter media: Media query for the resources intended media.
+        /// - parameter type: MIME type of the image, with optional `codecs` paramater.
+        /// - parameter height: Intrinsic height of image in pixels - integer with no unit.
+        /// - parameter width:  Intrinsic width of image in pixels - integer with no unit.
+
+        internal init(srcset: String, sizes: String? = nil, media: String? = nil, type: String? = nil, height: Int? = nil, width: Int? = nil) {
+            self.srcset = srcset
+            self.sizes = sizes
+            self.media = media
+            self.type = type
+            self.height = height
+            self.width = width
+        }
+    }
+
+    /// Initialize a `Picture` component with multiple `source` elements.
+    /// Image to use as fallback.
+    public var image: Plot.Image
+    /// A set of `source` elments from which a user agent will chose based on its requirements.
+    public var sources: [Source]?
+
+    internal init(image: Plot.Image, sources: [Picture.Source]? = nil) {
```

Comment:
Why does `Image` have to be prefixed with `Plot` here?

Reply:
Also, how about using an empty array as the default value, rather than an optional?

Reply:
Image was prefixed with Plot, as I developed this extension in my repository, not in a fork of Plot and Xcode/the compiler could not figure out what it was. :-) Now that it is in Plot, it is no longer needed and I have removed it.

Replaced it as you requested.